code/datums/gamemode/dynamic/dynamic_rulesets_latejoin.dm

```diff
@@ -197,6 +197,66 @@
 	newpd.AnnounceObjectives()
 	return 1
 
+//////////////////////////////////////////////
+//                                          //
+//                   GRUE                   ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/latejoin/grue
+	name = "Grue Infestation"
+	role_category = /datum/role/grue
+//	enemy_jobs
+//	required_enemies
+	required_candidates = 1
+	weight = BASE_RULESET_WEIGHT
+	cost = 25
+	requirements = list(5,5,15,15,20,20,20,20,40,70)
+	high_population_requirement = 10
+	logo = "grue-logo"
+
+	repeatable = TRUE
+	var/grue_spawn_spots=list()
+	var/turf/T
+/datum/dynamic_ruleset/latejoin/grue/ready(var/forced = 0)
+	grue_spawn_spots=list()
+	var/list/found_vents = list()
+	for(var/obj/machinery/atmospherics/unary/vent_pump/v in atmos_machines)
+		T=get_turf(v)
+		if(!v.welded && v.z == map.zMainStation && v.canSpawnMice==1&&T.get_lumcount()<=0.1&&istype(get_area(v),/area/maintenance)) // Same as spiders but with additional check for being in the dark and in maintenance.
+			found_vents.Add(v)
+	if(found_vents.len)
+		while(found_vents.len > 0)
+			var/obj/machinery/atmospherics/unary/vent_pump/v = pick(found_vents)
+			found_vents -= v
+			for (var/mob/M in player_list)
+				if (isliving(M) && (get_dist(M,v) > 7))//trying to find just one vent that is far out of view of any player
+					grue_spawn_spots+=get_turf(v)
+	if(!grue_spawn_spots)
+		log_admin("Cannot accept Grue ruleset, no suitable spawn locations found.")
+		message_admins("Cannot accept Grue ruleset, no spawn locations found.")
+		return 0
+
+	return ..()
+
+/datum/dynamic_ruleset/latejoin/grue/execute()
+	var/mob/M = pick(assigned)
+	var/turf/oldloc = get_turf(M)
+	M.forceMove(null)
+	if(!latejoinprompt(M,src))
+		message_admins("[M.key] has opted out of becoming a grue.")
```

Comment:
What would be a better alternative?

Reply:
This check should be in `ready` or something similar rather than in `execute`. Although I am not sure if a failed execute wastes the ruleset or not. I would have to double check. I'll get back.